chore(frontend): Lazy globals elaboration

vezenovm opened the pull request:
# Description

## Problem\*

Working towards green-lighting of the elaborator

## Summary\*

I was not seeing why we needed to split the elaboration of literal and non-literal globals in our elaborator. I think this may have been a remnant from our old name resolution strategy. It looks like using `elaborate_global_if_unresolved` inside of `resolve_named_type` has allowed for us to appropriately lazily resolve globals. 

## Additional Context

## Documentation\*

Check one:
- [X] No documentation needed.
- [x] Documentation included in this PR.
- [ ] **[For Experimental Features]** Documentation to be submitted in a separate PR.

# PR Checklist\*

- [X] I have tested the changes locally.
- [X] I have formatted the changes with [Prettier](https://prettier.io/) and/or `cargo fmt` on default settings.

vezenovm commented:
> This is probably not very important, just a curiosity I found.

Hmm this should compile. I thought this would be handled by the lazy globals elaboration but obviously things look like they are breaking. I will determine what is happening before merging this PR.

asterite commented:
I think the case that was working, and still working, is if you have `global bar: Foo<foo> = ...` where we reference `foo` in the type of the global. Then it's lazily elaborated and it works fine. But it breaks if `foo` is used in the value. It used to work before because literal values were elaborated before everything else.

vezenovm commented:
We were not appropriately lazy elaborating globals in the comptime interpreter. We were assuming that the global let statement was already elaborated. I switched to using `elaborate_global_if_unresolved` when we have an unresolved global value in the comptime interpreter. This also moves us to only use `globals` module for unresolved globals which is a good move either way. The failing test above now passes (and I added it as a frontend test). 

I could also use an approval on the parent PR https://github.com/noir-lang/noir/pull/10260 to get this change into master.
